doc/src/manual/performance-tips.md

```diff
@@ -1507,3 +1507,89 @@ The following examples may help you interpret expressions marked as containing n
         field `data::Array{T}`. But `Array` needs the dimension `N`, too, to be a concrete type.
       * Suggestion: use concrete types like `Array{T,3}` or `Array{T,N}`, where `N` is now a parameter
         of `ArrayContainer`
+
+## [Performance of captured variable](@id man-performance-captured)
+
+Consider the following example that defines two inner functions:
+```julia
+function abmult(r::Int)
+    if r >= 0
+        f = x -> x * r
+    else
+        r = -r
+        f = x -> x * r
+    end
+    return f
+end
```

Comment:
All these examples introduce two equal, but not identical closures, so even `abmult3` is not type-stable wrt. its return type. Can't you compare
```julia
function abmult(r::Int)
    if r < 0
        r = -r
    end
    return x -> x * r
end
```
with
```julia
function abmult3(r::Int)
    if r < 0
        r = -r
    end
    let r = r
        return x -> x * r
    end
end
```
to the same effect?

BTW, I really like the explanation of the problem here.